### DISCUSSION TITLE
IP Pool is not working, message always in Pending state

### DISCUSSION THREAD

**Discussion by maratgaliev** [postalserver/postal#3005] [Help with using Postal]
Hello.
I'm using AWS, this is my ip a command:

1: lo: <LOOPBACK,UP,LOWER_UP> mtu 65536 qdisc noqueue state UNKNOWN group default qlen 1000
    link/loopback 00:00:00:00:00:00 brd 00:00:00:00:00:00
    inet 127.0.0.1/8 scope host lo
       valid_lft forever preferred_lft forever
    inet6 ::1/128 scope host
       valid_lft forever preferred_lft forever
2: eth0: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 9001 qdisc mq state UP group default qlen 1000
    link/ether 06:c3:c4:af:dd:fd brd ff:ff:ff:ff:ff:ff
    inet 198.102.15.1/32 scope global eth0
       valid_lft forever preferred_lft forever
    inet 198.102.15.2/32 scope global eth0
       valid_lft forever preferred_lft forever
    inet 172.31.29.201/20 metric 100 brd 172.31.31.255 scope global dynamic eth0
       valid_lft 3574sec preferred_lft 3574sec
    inet6 fe80::4c3:c4ff:feaf:ddfd/64 scope link
       valid_lft forever preferred_lft forever
3: docker0: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1500 qdisc noqueue state UP group default
    link/ether 02:42:6e:db:96:56 brd ff:ff:ff:ff:ff:ff
    inet 172.17.0.1/16 brd 172.17.255.255 scope global docker0
       valid_lft forever preferred_lft forever
    inet6 fe80::42:6eff:fedb:9656/64 scope link
       valid_lft forever preferred_lft forever

This is my ip pool configuration, and I've added my 198.102.15.2 address:
![Screenshot 2024-06-11 at 11 12 26 AM](https://github.com/postalserver/postal/assets/190057/cb88e435-47ff-48af-aea9-425fe37a413b)

I also have RDNS records setup, dig shows me the right information

IP pools are enabled on Postal level, org level, mail server level, recreated everything multiple times.

I stuck on this for a couple weeks now and just don't understand what should I adjust/setup to make it work.

Thanks.

**Comment by maratgaliev** [comment]
Hey, any help?

**Reply by willpower232** [reply]
What is your `postal status`?

**Reply by maratgaliev** [reply]
NAME              IMAGE                               COMMAND                  SERVICE   CREATED      STATUS      PORTS
postal-smtp-1     ghcr.io/postalserver/postal:3.3.3   "/docker-entrypoint.…"   smtp      6 days ago   Up 2 days
postal-web-1      ghcr.io/postalserver/postal:3.3.3   "/docker-entrypoint.…"   web       6 days ago   Up 2 days
postal-worker-1   ghcr.io/postalserver/postal:3.3.3   "/docker-entrypoint.…"   worker    6 days ago   Up 2 days

**Reply by maratgaliev** [reply]
@willpower232 any ideas? I appreciate any help!

**Reply by willpower232** [reply]
> Created 6 days ago Up 2 days

Are you able to send new messages? Is it just messages sent before 2 days ago that are stuck?

**Reply by maratgaliev** [reply]
@willpower232 yes I can send emails with no pool enabled. If I enable the pool - state is always pending.

**Reply by willpower232** [reply]
It might be worth checking the logs of the workers to make sure they can connect to the IP addresses you have shown in the first comment.

**Reply by maratgaliev** [reply]
@willpower232 I have nothing in logs for the first minutes after sending message, pending and no even attempts to deliver.

But after some time I'm getting soft fail:

worker-1  | 2024-06-14 16:25:48 +0000 INFO   running task component=worker thread=tasks task=SendNotificationsScheduledTask
worker-1  | 2024-06-14 16:25:48 +0000 INFO   scheduling task to next run at 2024-06-14 16:26:48 UTC component=worker thread=tasks task=SendNotificationsScheduledTask
worker-1  | 2024-06-14 16:26:17 +0000 ERROR  Cannot connect to 142.251.165.27:25 (aspmx.l.google.com) (Net::OpenTimeout: execution expired) log_id=HK8ZY02L component=worker thread=work0 original_queued_message=21 queued_message=21
worker-1  | 2024-06-14 16:26:47 +0000 ERROR  Cannot connect to 172.253.62.27:25 (alt2.aspmx.l.google.com) (Net::OpenTimeout: execution expired) log_id=HK8ZY02L component=worker thread=work0 original_queued_message=21 queued_message=21
worker-1  | 2024-06-14 16:26:48 +0000 DEBUG  acquired task role by renewing it component=worker thread=tasks
worker-1  | 2024-06-14 16:26:48 +0000 INFO   running task component=worker thread=tasks task=SendNotificationsScheduledTask
worker-1  | 2024-06-14 16:26:48 +0000 INFO   scheduling task to next run at 2024-06-14 16:27:48 UTC component=worker thread=tasks task=SendNotificationsScheduledTask
worker-1  | 2024-06-14 16:27:17 +0000 ERROR  Cannot connect to 108.177.12.27:25 (alt1.aspmx.l.google.com) (Net::OpenTimeout: execution expired) log_id=HK8ZY02L component=worker thread=work0 original_queued_message=21 queued_message=21
worker-1  | 2024-06-14 16:27:47 +0000 ERROR  Cannot connect to 64.233.186.27:25 (alt3.aspmx.l.google.com) (Net::OpenTimeout: execution expired) log_id=HK8ZY02L component=worker thread=work0 original_queued_message=21 queued_message=21
worker-1  | 2024-06-14 16:27:48 +0000 DEBUG  acquired task role by renewing it component=worker thread=tasks
worker-1  | 2024-06-14 16:27:48 +0000 INFO   running task component=worker thread=tasks task=SendNotificationsScheduledTask
worker-1  | 2024-06-14 16:27:48 +0000 INFO   scheduling task to next run at 2024-06-14 16:28:48 UTC component=worker thread=tasks task=SendNotificationsScheduledTask
worker-1  | 2024-06-14 16:28:17 +0000 ERROR  Cannot connect to 209.85.202.26:25 (alt4.aspmx.l.google.com) (Net::OpenTimeout: execution expired) log_id=HK8ZY02L component=worker thread=work0 original_queued_message=21 queued_message=21
worker-1  | 2024-06-14 16:28:17 +0000 INFO   message requeued for trying later retry_after=2024-06-14 16:33:17 UTC component=worker thread=work0 original_queued_message=21 queued_message=2


<img width="779" alt="image" src="https://github.com/postalserver/postal/assets/190057/d1960a0a-6168-43f4-9605-3d9d6cd6d2f8">

**Reply by maratgaliev** [reply]
And this is actually weird, because my port 25 is open on AWS, no issues and with no pools enabled emails are sending fine.

**Reply by willpower232** [reply]
on AWS the public IP address isn't seen inside the server OS for voodoo networking so if you enable pools on AWS, you probably have to double check which IP address you enter into Postal as the IP.  Mine start with 172.31 and I can see which public IP that matches on the elastic IP page.

**Reply by maratgaliev** [reply]
Hey @willpower232 I really appreciate your help on this! Since you're on AWS, maybe you can help me with two questions:
- What type of actions should I complete on AWS/Ubuntu side to make these IPs available and usable with Postal? I have all of them added to the Elastic IP page
- How Postal internally works with that IP pool? Are these IP's just providing as parameter to some command?
Many thanks!

**Reply by maratgaliev** [reply]
@willpower232 hey! Sorry for bothering, any updates?
